server/server.js

```diff
@@ -1,21 +1,41 @@
 var express = require("express");
 var cookieParser = require("cookie-parser");
+var bodyParser = require("body-parser");
+var uuid = require("uuid");
+var _ = require("underscore");
 
-module.exports = function(port, db, githubAuthoriser) {
+module.exports = function (port, db, githubAuthoriser) {
     var app = express();
 
     app.use(express.static("public"));
     app.use(cookieParser());
+    app.use(bodyParser.json());
 
     var users = db.collection("users");
+    var conversations = db.collection("conversations");
     var sessions = {};
 
-    app.get("/oauth", function(req, res) {
-        githubAuthoriser.authorise(req, function(githubUser, token) {
+    var latestGuestID = 0;
+
+    function conversation(users) {
+        this._id = uuid.v4();
```

Comment:
Going to leave this here but it's a general comment. 
I am a bit concerned with your use of _id throughout. I assume you are using it because mongodb gives it's instances a unique _id. If this is the case, I feel you are tying your implementation to the use of mongo. 
I would let mongo keep it's own _id and create a new id for your entities.